### ISSUE TITLE
Shock Detector On/Off

### ISSUE THREAD

**Issue by DumbfireDecoy** [studio-minus/ppg-bugs-and-requests#2851]
### Feature description

Being able to toggle whether the Shock Detector sends a signal when impacting something. Preferably by sending a red/blue signal to the Shock Detector.

### Feature purpose

Would be a lot easier to make your own rockets/missiles that don't just spontaneously explode at every opportunity. Of course you could do something similar with logic gates, but the added signal delay from the components means the explosive goes off way too late after impacting, in addition to the increased amount of non-disintegrating parts that stick around after the round has gone off. 

In general, less of a pain in the ass to use Shock Detectors. :)

**Comment by Drogo-man** [comment]
Electric gate

**Comment by DumbfireDecoy** [comment]
> Electric gate

As I already stated, adding even just a single component delays the signal too much. Also it's generally nice to reduce the number of objects you need to achieve what you want.

**Comment by Squibard** [comment]
> Of course you could do something similar with logic gates, but the added signal delay from the components means the explosive goes off way too late after impacting

"Of course you could do something similar with logic gates, but the added signal delay from the components means the explosive goes off way too late after impacting"

**Comment by Squibard** [comment]
> Electric gate

"Of course you could do something similar with logic gates, but the added signal delay from the components means the explosive goes off way too late after impacting"

**Comment by WWWOWWWPL** [comment]
I am pretty sure the propagation system is getting reworked soon, so the delay problem might disappear eventually

**Comment by DumbfireDecoy** [comment]
> I am pretty sure the propagation system is getting reworked soon, so the delay problem might disappear eventually

Yeah the rework would solve it, but just the possibilty of disabling the shock detector would reduce part count, which is always a plus.
